### DISCUSSION TITLE
How to auto refresh grid every n seconds?

### DISCUSSION THREAD

**Discussion by juliermefelix** [serenity-is/Serenity#6277] [Help]
How to auto refresh grid every n seconds?

I tried to update the .cshtml file, but when I open a dialog and the refresh page I lose the dialog data, so I need an alternative that only does it on the grid page, including the code in grid.ts

code used in .cshtml

`  <script type="text/javascript">
             setTimeout(function() {
                 location.reload();
             }, 60 * 1000);
         </script>`

**Comment by minhhungit** [comment]
@juliermefelix try put this into your grid.ts file

```typescript
constructor(container: JQuery) {
    super(container);

    setTimeout(() => {
        setInterval(() => {
            this.refresh();
        }, 2 * 1000); // refresh every 2 seconds
    }, 5 * 1000); // delay 5 seconds after first loading
          
}
```
